akka-javasdk/src/main/java/akka/javasdk/agent/Agent.java

```diff
@@ -67,8 +67,20 @@ interface Builder {
 
       Builder systemMessage(String message);
 
+      /**
+       * Create a system message from a template. Call @{@link PromptTemplate} before to initiate or update template value.
+       * @param templateId the id of the template to use
+       */
       Builder systemMessageFromTemplate(String templateId);
 
+      /**
+       * Create a system message from a template. Call @{@link PromptTemplate} before to initiate or update template value.
+       * Provide arguments that will be applied to the template using Java {@link String#format} method.
+       * @param templateId the id of the template to use
+       * @param args the arguments to apply to the template
+       */
+      Builder systemMessageFromTemplate(String templateId, Object... args);
```

Comment:
can we later support named parameters with an overloaded method taking a Map parameter, or will that collide with this one?

Reply:
it will, we could do the runtime check if the first parameter is map

Reply:
users can always use componentClient and to any formatting they want

Reply:
> we could do the runtime check if the first parameter is map

I think that's good enough, if we do that later

Reply:
we can also add some variants like we do for persisting events:
```java
systemMessageFromTemplate(String templateId, Object a);
systemMessageFromTemplate(String templateId, Object a1, Object a2);
systemMessageFromTemplate(String templateId, Object a1, Object a2, Object... args);
```

Then, only the first can receive a Map. Easier to check if it was intentional or if people are mixing Map with other regular args.

Reply:
We should be able to check such things anyway. In the end it's just an array.

Reply:
specialized methods for the most common number of args could have a small performance advantage since it doesn't create the intermediate array, but we are not at that level of performance optimization here